cmd/nex/monitor.go

```diff
@@ -1,10 +1,122 @@
 package main
 
+import (
+	"fmt"
+	"log/slog"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/synadia-io/nex/api/nodecontrol"
+	"golang.org/x/net/context"
+)
+
 type Monitor struct {
 	Events Events `cmd:"" help:"Monitor events"`
 	Logs   Logs   `cmd:"" help:"Monitor logs"`
 }
 
 // Monitor subcommands
-type Events struct{}
-type Logs struct{}
+type Events struct {
+	WorkloadID string `default:"*" help:"Workload ID to filter logs"`
+	EventType  string `default:"*" help:"Filter events on type"`
+}
+
+func (e *Events) Run(ctx context.Context, globals *Globals) error {
+	if globals.Check {
+		return printTable("Monitor Events Configuration", append(globals.Table(), e.Table()...)...)
+	}
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+	go func() {
+		<-c
+		cancel()
+	}()
+
+	nc, err := configureNatsConnection(globals)
+	if err != nil {
+		return err
+	}
+
+	controller, err := nodecontrol.NewControlApiClient(nc, slog.New(slog.NewTextHandler(os.Stdin, nil)))
+	if err != nil {
+		return err
+	}
+
+	logs, err := controller.MonitorEvents(e.WorkloadID, e.EventType)
+	if err != nil {
+		return err
+	}
+
+	eWord := "All"
+	if e.EventType != "*" {
+		eWord = e.EventType
+	}
+	fmt.Printf("##### %s event types for application: %s\n", eWord, e.WorkloadID)
+
+logloop:
+	for {
+		select {
+		case <-ctx.Done():
+			break logloop
```

Comment:
Would a continue work here instead of a labeled break?

Reply:
it was originally like that, but I think it continued the select instead of breaking out of the for....I cant remember